scripts/install_poetry.py

```diff
@@ -40,11 +36,12 @@
 from pathlib import Path
 import re
 import shutil
+import ssl
 import subprocess
 import sysconfig
 import tempfile
 from urllib.request import Request, urlopen
-from typing import Optional
+from typing import Generator, Optional
```

Comment:
Most of these changes are mypy appeasement. Since this is a script we expect to maintain, it seemed worth doing